ehttp/src/types.rs

```diff
@@ -93,8 +93,31 @@ impl<'h> IntoIterator for &'h Headers {
 
 // ----------------------------------------------------------------------------
 
+/// Determine if cross-origin requests lead to valid responses.
+#[cfg(target_arch = "wasm32")]
+#[derive(Default, Clone, Copy, Debug)]
+pub enum Mode {
+    SameOrigin = 0,
+    NoCors = 1,
+    #[default]
+    Cors = 2,
+    Navigate = 3,
```

Comment:
Please add some docstrings to these alternatives, perhaps with a reference to https://developer.mozilla.org/en-US/docs/Web/API/Request/mode